api/monitor/load.go

```diff
@@ -0,0 +1,123 @@
+package monitor
+
+import (
+	"encoding/json"
+	"github.com/arkhn/fhir-river/api/routes/batch"
+	"github.com/arkhn/fhir-river/api/topics"
+	"github.com/confluentinc/confluent-kafka-go/kafka"
+	"github.com/go-redis/redis"
+	"log"
+	"os"
+	"os/signal"
+	"strconv"
+	"syscall"
+)
+
+type message struct {
+	BatchID    string `json:"batch_id"`
+	ResourceID string `json:"resource_id"`
+}
+
+func isEndOfBatch(msg message, rdb *redis.Client) (bool, error) {
+	batchResources, err := rdb.SMembers("batch:"+msg.BatchID+":resources").Result()
+	if err != nil {
+		return false, err
+	}
+	counter, err := rdb.HGetAll("batch:"+msg.BatchID+":counter").Result()
+	if err != nil {
+		return false, err
+	}
+	for _, resource := range batchResources {
+		extractCount, ok := counter["resource:"+resource+":extract"]
+		if !ok {
+			return false, nil
+		}
+		count, err := strconv.ParseInt(extractCount, 10, 32)
```

Comment:
What are 10 and 32? Should the be hardcoded here?

Reply:
Base 10, 32 bits. I think it's ok to let it this way.